student.rb

```diff
@@ -0,0 +1,14 @@
+class Student < Person
+  def initialize(name, classroom:, parent_permission: true, age: 'Unknown')
+    @parent_permission = parent_permission
+    @name = name
+    @age = age
+    @classroom = classroom
+    @id = 0
+    super
+  end
```

Comment:
Let's refactor this `initialize` method to meet the inheritance requirements:

- [ ] Let's notice that your `super` class is expecting three params: `name`, `age`, and `parent_permission`, if you look closely, you're calling your `super` without any params, which is going to end up in an argument error. Instead, please consider calling your `super` class with its needed arguments:

```rb
  def initialize(name, classroom:, parent_permission: true, age: 'Unknown')
    super(name, age, parent_permission: parent_permission)
    @classroom = classroom
  end
```
 This way, you're calling the `super` method to create a person and then adding the extra properties that are specific to a `student`. In this example, `classroom`. 🤓 🔧